source-map.bs

```diff
@@ -245,17 +251,19 @@ Resolving Sources {#resolving-sources}
 If the sources are not absolute URLs after prepending the [=sourceRoot=], the sources are
 resolved relative to the SourceMap (like resolving the script `src` attribute in an HTML document).
 
-Encoding {#encoding}
---------------------
-
-For simplicity, the character set encoding is always UTF-8.
+Fetching Source Maps {#fetching-source-maps}
+--------------------------------------------
 
-Compression {#compression}
---------------------------
+To fetch a source map, given a [=URL=] |url|, run the following steps:
 
-The file is allowed to be GZIP compressed.   It is not expected that in-browser consumers of
-the source map will support GZIP compression directly but that they will consume an
-uncompressed map that may be GZIP'd for transport.
+1. Let |promise| be [=a new promise=].
+1. Let |request| be a new [=request=] whose [=request/URL=] is |url|.
+1. [=Fetch=] |request| with [=processResponseConsumeBody=] set to the following steps given [=response=] <var ignore>response</var> and null, failure, or a [=byte sequence=] |bodyBytes|:
+    1. If |bodyBytes| is null or failure, [=reject=] |promise| with a {{TypeError}} and abort these steps.
+    1. Let |sourceMap| be the result of [=parsing JSON bytes to a JavaScript value=] given |bodyBytes|.
```

Comment:
This is basically `JSON.parse(utf8decode(bodyBytes))`. There is also the option of returning an "internal spec value" rather than a JS object, but our spec is so high level that it doesn't really matter.